cadc-vosi/src/main/java/ca/nrc/cadc/vosi/Availability.java

```diff
@@ -84,113 +84,121 @@
 
 /**
  * @author zhangsa
- *
  */
-public class Availability
-{
+public class Availability {
     private AvailabilityStatus _status;
 
-    public Availability(AvailabilityStatus status)
-    {
+    public Availability(AvailabilityStatus status) {
         super();
-        if (status == null)
+        if (status == null) {
             throw new IllegalArgumentException("Availability Status is null.");
+        }
         _status = status;
     }
-    
-    public Availability(Document xml)
-    {
+
+    public Availability(Document xml) {
         super();
-        if (xml == null)
+        if (xml == null) {
             throw new IllegalArgumentException("Document is null.");
-        try
-        {
-            _status = fromXmlDocument(xml);
         }
-        catch (ParseException e)
-        {
+        try {
+            _status = fromXmlDocument(xml);
+        } catch (ParseException e) {
             throw new IllegalStateException("Invalid date format in XML", e);
         }
     }
 
-    public Document toXmlDocument()
-    {
+    public Document toXmlDocument() {
         DateFormat df = DateUtil.getDateFormat(DateUtil.IVOA_DATE_FORMAT, DateUtil.UTC);
         Namespace vosi = Namespace.getNamespace("vosi", VOSI.AVAILABILITY_NS_URI);
 
         Element eleAvailability = new Element("availability", vosi);
-        
+
         Util.addChild(eleAvailability, vosi, "available", Boolean.toString(_status.isAvailable()));
-        if (_status.getUpSince() != null)
+        if (_status.getUpSince() != null) {
             Util.addChild(eleAvailability, vosi, "upSince", df.format(_status.getUpSince()));
-        if (_status.getDownAt() != null)
+        }
+        if (_status.getDownAt() != null) {
             Util.addChild(eleAvailability, vosi, "downAt", df.format(_status.getDownAt()));
-        if (_status.getBackAt() != null)
+        }
+        if (_status.getBackAt() != null) {
             Util.addChild(eleAvailability, vosi, "backAt", df.format(_status.getBackAt()));
-        if (_status.getNote() != null)
+        }
+        if (_status.getNote() != null) {
             Util.addChild(eleAvailability, vosi, "note", _status.getNote());
+        }
+        if (_status.getClientIP() != null) {
+            eleAvailability.addContent(new Comment(String.format("<clientip>%s</clientip>", _status.getClientIP())));
+        }
```

Comment:
See comment in AvailabilityStatus.java -- this method should get the IP directly from NetUtil.getClientIP().